File: scripts/remove-logo-background.mjs
/**
 * Make white background of RDT_Logo.png transparent.
 * Run: node scripts/remove-logo-background.mjs
 */
const Jimp = (await import("jimp")).default;

const LOGO_PATH = "public/RDT_Logo.png";
const WHITE_THRESHOLD = 250;

async function main() {
  const img = await Jimp.read(LOGO_PATH);
  if (img.hasAlpha()) {
    // ensure we have alpha channel to modify
  } else {
    img.rgba(false);
  }

  img.scan(0, 0, img.bitmap.width, img.bitmap.height, function (x, y, idx) {
    const r = this.bitmap.data[idx];
    const g = this.bitmap.data[idx + 1];
    const b = this.bitmap.data[idx + 2];
    if (r >= WHITE_THRESHOLD && g >= WHITE_THRESHOLD && b >= WHITE_THRESHOLD) {
      this.bitmap.data[idx + 3] = 0;
    }
  });

  await img.writeAsync(LOGO_PATH);
  console.log("Saved", LOGO_PATH, "with transparent background.");
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: scripts/remove-logo-background.py
"""
Make white background of RDT_Logo.png transparent.
Requires: pip install Pillow
"""
from PIL import Image

LOGO_PATH = "public/RDT_Logo.png"

def main():
    img = Image.open(LOGO_PATH).convert("RGBA")
    data = img.getdata()
    new_data = []
    # Threshold: pixels this close to white become transparent
    threshold = 250
    for item in data:
        r, g, b, a = item
        if r >= threshold and g >= threshold and b >= threshold:
            new_data.append((255, 255, 255, 0))
        else:
            new_data.append(item)
    img.putdata(new_data)
    img.save(LOGO_PATH)
    print(f"Saved {LOGO_PATH} with transparent background.")

if __name__ == "__main__":
    main()
